### ISSUE TITLE
Toggling/muting/unmuting target with --device flag requires --source flag (for sources), but not --sink (for sinks)

### ISSUE THREAD

**Issue by SuperFluffy** [falconindy/ponymix#31]
When directly targetting a device with the `--device` flag to toggle its state (or muting/unmuting it), the `--source` flag is required for sources, but the `--sink` flag is not equivalently required for sinks. Demonstration:
```
# For the output / sink device
% ponymix -d alsa_output.pci-0000_00_1b.0.analog-stereo toggle
85

# For the input / source device
% ponymix -d alsa_input.pci-0000_00_1b.0.analog-stereo toggle 
ponymix: no match found for device: alsa_input.pci-0000_00_1b.0.analog-stereo
% ponymix --source -d alsa_input.pci-0000_00_1b.0.analog-stereo toggle
36
```

**Comment by falconindy** [comment]
What are you suggesting as a better behavior? This Just Works™ for sinks because I've decided that devtype should default to "sink" -- this is generally what you want because it controls output volume (what I consider to be the most common usecase). Removing the default means command lines get more verbose.

**Comment by SuperFluffy** [comment]
Yeah, I was just looking at the source, and noticed that `opt_devtype = DEVTYPE_SINK;` in `main()`. I guess the structure of `GetDevice` in `pulse.cc` is dictated by the PulseAudio API (through `get_device`)?

Otherwise I'd suggest that if the `-d DEVICE` flag is set, and neither `--source` nor `--sink` are specified, that the device type should be infered via its name? I am not really sure if they API is there, but since the devices are classified as either sinks or sources, and since the devices are uniquely identified by their names, this should be possible?

**Comment by falconindy** [comment]
Makes sense. I don't know of any way to infer device type directly from the name, and the pulse API doesn't offer this idea of a "blind" lookup either. However, it does seem like we could just loop over the internal vectors of devices in the PulseClient until we find a match. Interested in writing a patch?

**Comment by SuperFluffy** [comment]
Yeah, I will look at it. This will help me brush up my C++.

Though this will take me a bit of time, if that's alright with you. :)

**Comment by falconindy** [comment]
Fine by me! Thanks!

**Comment by SuperFluffy** [comment]
This turned out to be pretty straight forward. `--source` and `--sink` are still useful, although I not completely happy with them being sillently ignored when supplying `-d DEVICE`.
